src/runtime/StateSerialization/MaybeMemberInfo.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.IO;
+
+namespace Python.Runtime
+{
+     [Serializable]
+    internal struct MaybeMemberInfo<T> : ISerializable where T: MemberInfo
+    {
+        public static implicit operator MaybeMemberInfo<T> (T ob) => new MaybeMemberInfo<T>(ob);
+
+        // .ToString() of the serialized object
+        const string SerializationName = "s";
+        // The ReflectedType of the object
+        const string SerializationType = "t";
+        const string SerializationFieldName = "f";
+        string name;
+        MemberInfo info;
+
+        [NonSerialized]
+        Exception deserializationException;
+
+        public string DeletedMessage 
+        {
+            get
+            {
+                return $"The .NET {typeof(T)} {name} no longer exists. Cause: " + deserializationException?.Message ;
+            }
+        }
+
+        public T Value
+        {
+            get
+            {
+                if (info == null)
+                {
+                    throw new SerializationException(DeletedMessage, innerException: deserializationException);
+                }
+                return (T)info;
+            }
+        }
+
+        public string Name {get{return name;}}
+        public bool Valid => info != null;
+
+        public override string ToString()
+        {
+            return (info != null ? info.ToString() : $"missing type: {name}");
+        }
+
+        public MaybeMemberInfo(T fi)
+        {
+            info = fi;
+            name = info?.ToString();
```

Comment:
Name isn't really required now. You only need it if info is null.

You need to always serialize the name; if info is set, then serialize info.ToString else serialize name.
You only need to deserialize the name if info failed to deserialize for whatever reason.

Reply:
Maybe it doesn't matter, since the memory savings aren't huge, and debugging is easier if you always have the name.

Reply:
> debugging is easier if you always have the name.

That's why I always have it